include/wtr/watcher.hpp

```diff
@@ -944,6 +945,18 @@ inline auto is_dir(char const* const path) -> bool
   return stat(path, &s) == 0 && S_ISDIR(s.st_mode);
 }
 
+inline auto should_skip =
+  [](
+    char const* const dir,
+    std::vector<std::string> const& ignored_paths) -> bool
+{
+  return std::any_of(
+    ignored_paths.begin(),
+    ignored_paths.end(),
+    [&](std::string const& ignored)
+    { return strcmp(dir, ignored.c_str()) == 0; });
+};
+
```

Comment:
suggestion: Here, and elsewhere throughout the PR, let's use `std::set<std::string> ignored_paths` to represent this. Reasoning is that constant-time lookups seem like the play here. In most cases, the performance difference won't matter at all, but if someone comes along with thousands of items in here, then we have to make an API-breaking change to better accommodate that.